asserts/fetcher.go

```diff
@@ -175,6 +175,9 @@ func (f *fetcher) fetchSequence(seq *AtSequence) error {
 
 // FetchSequence retrieves the assertion as indicated by its sequence reference.
 func (f *fetcher) FetchSequence(seq *AtSequence) error {
+	if f.retrieveSeq == nil {
```

Comment:
~~While I agree with the change I'm curious what prompted this? The type system should protect us from reaching this point unless playing shenanigans with type assertions (like you did in the tests)? Am I missing something? (fine obviously to be definitive here, just curious)~~ 

Answered myself below